### PR TITLE
NEXUS-8287 NuGet api key ui

### PR DESCRIPTION
**Pull request by kellyrob99**
- utilizes existing UI elements from nexus-nuget-plugin and moves the functionality to the user feature in the UI

Present nexus pro UI for this functionality adds a tab to each nuget repo: http://screencast.com/t/3iGWR6aBQE
Proposed change puts this under the User's account management: http://screencast.com/t/0AFKmHZsDucV

### CODE REVIEW

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java [hunk 1]
```diff
@@ -62,6 +62,12 @@ public NugetApiKeyStoreImpl(final @Named("security") Provider<DatabaseInstance>
     this.databaseInstance = checkNotNull(databaseInstance);
     this.entityAdapter = checkNotNull(entityAdapter);
     this.principalsHelper = checkNotNull(principalsHelper);
+    try {
```
Comment (kellyrob99) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java]: need a place to start this from, @all is there a common paradigm for initialization or simply EventBus+NexusStartedEvent on a lifecycle component to bootstrap this?

File: plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/app/PluginConfig.js [hunk 2]
```diff
@@ -121,9 +121,9 @@ Ext.define('NX.coreui.app.PluginConfig', {
         return NX.app.Application.pluginActive('org.sonatype.nexus.plugins:nexus-coreui-plugin');
       }
     },
-    { id: 'NX.coreui.controller.NuGetRepositorySettings',
+    { id: 'NX.coreui.controller.NuGetApiKey',
       active: function () {
-        return NX.app.Application.pluginActive('com.sonatype.nexus.plugins:nexus-nuget-plugin');
+        return NX.app.Application.pluginActive('org.sonatype.nexus.plugins:nexus-coreui-plugin');
```
Comment (kellyrob99) [inline plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/app/PluginConfig.js]: moved from the nexus-nuget-plugin, which brings up the question: Can we delete this plugin now that the functionality is all in OSS?

Comment (mrprescott) [inline plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/app/PluginConfig.js]: That's the plan - I opened https://issues.sonatype.org/browse/NEXUS-8355 to remind myself/us.

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java (hunk 1 above)
Comment (jdillon) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java]: using events right now is the main way to trigger lifecycle.

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java [hunk 3]
```diff
@@ -62,6 +62,12 @@ public NugetApiKeyStoreImpl(final @Named("security") Provider<DatabaseInstance>
...
+      start();
+    }
+    catch (Exception e) {
+      e.printStackTrace();
```
Comment (jdillon) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java]: never use this in production code rethrow or log.

also not a good idea to start on ctor.

Comment (kellyrob99) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyStoreImpl.java]: agreed and @mrprescott and I talked about something similar to RepositoryLifecycle to bootstrap this

**Pull request merged**